### PR TITLE
Add comments for VERSION_REVISION vs DBMS_TCP_PROTOCOL_VERSION

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Cc: @akuzm

### CODE REVIEW

File: src/Core/Defines.h [hunk 1]
```diff
@@ -85,7 +85,13 @@
 
 #define DBMS_MIN_PROTOCOL_VERSION_WITH_DISTRIBUTED_DEPTH 54448
 
-/// Version of ClickHouse TCP protocol. Increment it manually when you change the protocol.
+/// Version of ClickHouse TCP protocol.
+///
+/// Should be incremented it manually on protocol changes.
```
Comment (alesapin) [inline src/Core/Defines.h:90]:
```suggestion
/// Should be incremented manually on protocol changes.
```

**Review by alesapin — APPROVED** [review]

File: src/Core/Defines.h (hunk 1 above)
Comment (azat) [reply src/Core/Defines.h]: Applied, thanks!

**Comment by alesapin** [comment]
No related failures.

**Comment by azat** [comment]
>[ClickHouse Fast Test for PR #25771](https://clickhouse-test-reports.s3.yandex.net/25771/dfbc80f08a5130c88a235b69366b858772ff572e/fast_test.html#fail1)

`01532_execute_merges_on_single_replica`

<details>

```
2021-06-28 10:58:37 01532_execute_merges_on_single_replica:                                 [ FAIL ] - result differs with reference:
2021-06-28 10:58:37 --- /ClickHouse/tests/queries/0_stateless/01532_execute_merges_on_single_replica.reference	2021-06-28 10:53:57.311977299 +0300
2021-06-28 10:58:37 +++ /ClickHouse/tests/queries/0_stateless/01532_execute_merges_on_single_replica.stdout	2021-06-28 10:58:37.387104983 +0300
2021-06-28 10:58:37 @@ -6,14 +6,6 @@
2021-06-28 10:58:37  *** timeout not exceeded, r1 waits for r2
2021-06-28 10:58:37  Row 1:
2021-06-28 10:58:37  ──────
2021-06-28 10:58:37 -table:           execute_on_single_replica_r1
2021-06-28 10:58:37 -type:            MERGE_PARTS
2021-06-28 10:58:37 -new_part_name:   all_0_0_5
2021-06-28 10:58:37 -has_postpones:   1
2021-06-28 10:58:37 -postpone_reason: Not executing merge for the part all_0_0_5, waiting for r2 to execute merge.
2021-06-28 10:58:37 -
2021-06-28 10:58:37 -Row 2:
2021-06-28 10:58:37 -──────
2021-06-28 10:58:37  table:           execute_on_single_replica_r2
2021-06-28 10:58:37  type:            MERGE_PARTS
2021-06-28 10:58:37  new_part_name:   all_0_0_5
```

</details>

The server was too slow:

<details>

```2021.06.28 10:58:11.743849 [ 20348 ] {4ad7ba0d-e3f3-4fb0-b84f-1c655842c5df} <Debug> executeQuery: (from [::1]:47580, using production parser) (comment: 01532_execute_merges_on_single_replica.sql) /* all_0_0_5 - should be merged by r2, but it has replication queue stopped, >
 OPTIMIZE TABLE execute_on_single_replica_r1 FINAL SETTINGS replication_alter_partitions_sync=0;
...
2021.06.28 10:58:11.747104 [ 20176 ] {} <Debug> test_5d7c3x.execute_on_single_replica_r1 (ReplicatedMergeTreeQueue): Pulled 1 entries to queue.
2021.06.28 10:58:11.748210 [ 20175 ] {} <Debug> test_5d7c3x.execute_on_single_replica_r2 (ReplicatedMergeTreeQueue): Pulled 1 entries to queue.
... 14 seconds gap ...
2021.06.28 10:58:25.583104 [ 20262 ] {} <Trace> test_5d7c3x.execute_on_single_replica_r1 (StorageReplicatedMergeTree::mutationsFinalizingTask): Execution took 2947 ms.
2021.06.28 10:58:25.583137 [ 20174 ] {} <Trace> test_5d7c3x.execute_on_single_replica_r1 (StorageReplicatedMergeTree::mergeSelectingTask): Execution took 3842 ms.
...
2021.06.28 10:58:25.583133 [ 22064 ] {} <Trace> test_5d7c3x.execute_on_single_replica_r1 (0569d08b-415c-4019-8569-d08b415c0019): Executing log entry to merge parts all_0_0_4 to all_0_0_5
```

</details>

So I guess increasing `execute_merges_on_single_replica_time_threshold` may fix the test, but it is already high enough.
By the way @alesapin maybe logs should be moved into separate filesystem or tmpfs? Then these random latencies can go away I guess.

*I wish fasttest has exported trace_log, then we can see why it takes so long, likely so fsync or similar*

Cc: @filimonov

**Pull request merged**